src/ng/q.js

```diff
@@ -254,25 +254,34 @@ function qFactory(nextTick, exceptionHandler) {
    *
    * @returns {Deferred} Returns a new instance of deferred.
    */
-  var defer = function() {
-    var d = new Deferred();
+  var defer = function(keepTrack) {
+    var d = new Deferred(keepTrack);
     //Necessary to support unbound execution :/
     d.resolve = simpleBind(d, d.resolve);
     d.reject = simpleBind(d, d.reject);
     d.notify = simpleBind(d, d.notify);
     return d;
   };
 
-  function Promise() {
+  function Promise(keepTrack) {
     this.$$state = { status: 0 };
+
+    // some built-in angular module may use promises when the dependencies are not yet loaded, make sure not to track those
+    if (keepTrack || promiseTracker) {
+      this.keepTrack = keepTrack !== undefined ? keepTrack : true;
+    }
+  
+    if (this.keepTrack === true) {
+      promiseTracker.track(this);
+    }
   }
 
   extend(Promise.prototype, {
     then: function(onFulfilled, onRejected, progressBack) {
       if (isUndefined(onFulfilled) && isUndefined(onRejected) && isUndefined(progressBack)) {
         return this;
       }
-      var result = new Deferred();
+      var result = new Deferred(false); // do not track the .then() as separate promises
```

Comment:
Why not?

Reply:
In the initial implementation we figured cancelling a parent promise would cancel its children. Let me know if this is an issue